### PR TITLE
Convert FooterBlock into RSC

### PR DESCRIPTION
**Pull request by alebedev**
## Describe your changes

- Extract LanguageSelectForm from FooterBlock
- Convert FooterBlock styles to vanilla extract
- Remainder of FooterBlock is now a server component
- Always use next/link, we had some dead code swapping components between internal and external links
- Disable prefetch on footer links, they aren't interacted with often enough

## Justify why they are needed

Make FooterBlock easier to change in the future

## Checklist before requesting a review

- [ ] I have performed a self-review of my code

### CODE REVIEW

**Comment by alebedev** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/4389?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#4389** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/4389?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#4387** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/4387?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @alebedev and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: apps/store/src/blocks/FooterBlock.tsx [hunk 1]
```diff
@@ -1,160 +0,0 @@
...
-import { useCurrentLocale } from '@/utils/l10n/useCurrentLocale'
-import { disclaimerStyle, wrapperStyle } from './FooterBlock.css'
-
-type FooterLinkProps = SbBaseBlockProps<{
-  link: LinkField
-  linkText: string
-}>
-
-export const FooterLinkBlock = ({ blok }: FooterLinkProps) => {
-  const Component = blok.link.linktype === 'url' ? StyledAnchor : StyledLink
```
Comment (alebedev) [inline apps/store/src/blocks/FooterBlock.tsx:29]: This was effectively dead code, we're rendering same HTML anyway

File: apps/store/src/blocks/FooterBlock/FooterBlock.tsx [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+
+type FooterSectionProps = SbBaseBlockProps<{
+  footerLinks: ExpectedBlockType<FooterLinkProps>
+  title: string
+}>
+
+export const FooterSectionBlock = ({ blok }: FooterSectionProps) => {
+  const filteredFooterLinks = filterByBlockType(blok.footerLinks, FooterLinkBlock.blockName)
+  return (
+    <Space y={1.5} {...storyblokEditable(blok)}>
```
Comment (gustaveen) [inline apps/store/src/blocks/FooterBlock/FooterBlock.tsx:46]: nit: want to replace with stack? :)

File: apps/store/src/blocks/FooterBlock/LanguageSelectForm.tsx
Comment (gustaveen) [inline apps/store/src/blocks/FooterBlock/LanguageSelectForm.tsx:1]: Nice that you made it it's own component 👌

**Review by gustaveen — APPROVED** [review]

File: apps/store/src/blocks/FooterBlock/FooterBlock.tsx (hunk 2 above)
Comment (alebedev) [reply apps/store/src/blocks/FooterBlock/FooterBlock.tsx:46]: It's in next PR :)

**Comment by alebedev** [comment]
### Merge activity

* **Jun 20, 4:35 AM EDT**: [Graphite](https://app.graphite.dev/github/pr/HedvigInsurance/racoon/4389) rebased this pull request after merging its parent, because this pull request is set to merge when ready.

**Pull request merged**